shared/utils/replaceLinks.ts

```diff
@@ -1,9 +1,11 @@
 // https://stackoverflow.com/a/3809435
+// [^\u0020-\u007F] matches non-ASCII characters (e.g. umlaute, cyrillic, arabic)
+// https://stackoverflow.com/a/150078
 const protocol = /([A-Za-z]{3,9}:(?:\/\/)?)/
 const hostname = /(?:[-;:&=+$,\w]+@)?([A-Za-z0-9-]+(?:\.[A-Za-z0-9-]+)*)/
-const path = /(?:\/(?:[+~%/.\w-]*[+~%/\w-]+)?)?/
-const query = /(?:\?([-+=&;%@.\w]*[-+=&;%@\w]+)?)?/
-const hash = /(?:#([.!/\\\w]*[!/\\\w]+)?)?/
+const path = /(?:\/(?:([+~%/.\w-]|[^\u0020-\u007F])*([+~%/\w-]|[^\u0020-\u007F]+))?)?/
```

Comment:
This does not really make sense. You say it should be a word with latin leters and +~... or it should match anything. The second part is always true, when the first part is true, so the first part can be deleted.

Also i do not get why this is repeated once with the * and once with the +. Coudn't the * part be deleted as it is just duplication with no changes?

Reply:
@ztefanie it does make sense. E.g. for path segemnts, dots (`.`) are allowed in the middle of a segment but not at the end, thats why there is **nearly** identical  parts in the regexes.

But I adjusted the unicode matching part according to @sarahsporck's suggestion :)

Reply:
Sorry about not being clear, with first part, i meant `[+~%/.\w-]` (the first part before the or) and by second part i meant `[^\u0020-\u007F]`. 

> You say it should be a word with latin leters and +~... or it should match anything. The second part is always true, when the first part is true, so the first part can be deleted.

Reply:
Ahhh okay. Thats not the case anymore though, is it?